pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -509,15 +525,15 @@ func (gce *GCECloud) deleteForwardingRule(name, region string) error {
 }
 
 // DeleteTargetPool deletes the given target pool.
-func (gce *GCECloud) DeleteTargetPool(name string, hc *compute.HttpHealthCheck) error {
+func (gce *GCECloud) DeleteTargetPool(name string, hcNames ...string) error {
 	region, err := GetGCERegion(gce.localZone)
 	if err != nil {
 		return err
 	}
-	return gce.deleteTargetPool(name, region, hc)
+	return gce.deleteTargetPool(name, region, hcNames...)
 }
 
-func (gce *GCECloud) deleteTargetPool(name, region string, hc *compute.HttpHealthCheck) error {
+func (gce *GCECloud) deleteTargetPool(name, region string, hcNames ...string) error {
```

Comment:
Just to write down what we talked about offline, we could delete the health check that's linked from the targetpool. I suppose it's trading one DELETE (one of the health checks) call for one GET call (target pool). I don't know if it buys us anything other than possible code cleanliness. 

Just an idea for you to explore - up to you on whether we go with it.